src/app/pages/about-us/components/feedback-form/feedback-form.component.ts

```diff
@@ -0,0 +1,67 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, OnInit } from '@angular/core';
+import { FormGroup, FormControl, Validators, AbstractControl } from '@angular/forms';
+import { FeedbackFacadeService } from '@shared-module';
+
+@Component({
+  selector: 'app-feedback-form',
+  templateUrl: './feedback-form.component.html',
+  styleUrls: ['./feedback-form.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class FeedbackFormComponent implements OnInit {
+  feedbackForm: FormGroup;
+  currentRate: number;
+  isThankYouMessageVisible = false;
+  private readonly emailRegex = /^([a-zA-Z0-9._%-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,})$/;
+  private readonly delayTime = 3000;
+
+  constructor(private feedbackFacadeService: FeedbackFacadeService, private cdr: ChangeDetectorRef) {}
+
+  ngOnInit(): void {
+    this.feedbackForm = new FormGroup({
+      feedbackMessage: new FormControl(null, Validators.required),
+      name: new FormControl(null, Validators.required),
+      email: new FormControl(null, [Validators.required, Validators.pattern(this.emailRegex)]),
+    });
+  }
+
+  buildTranslationKey(relativeKey: string): string {
+    return `aboutUsPage.feedbackForm.${ relativeKey }`;
+  }
+
+  onSubmit(): void {
+    const newFeedback = { ...this.feedbackForm.value, rate: this.currentRate };
+    this.feedbackFacadeService.createNewFeedback(newFeedback);
+    this.feedbackForm.reset();
+    this.isThankYouMessageVisible = true;
+
+    setTimeout(() => {
+      this.isThankYouMessageVisible = false;
+      this.cdr.markForCheck();
+    }, this.delayTime);
+  }
+
+  newCurrentRate(currentRate: number): void {
+    this.currentRate = currentRate;
+  }
+
+  getControl(controlName: string): AbstractControl {
+    return this.feedbackForm.get(controlName);
+  }
+
+  isTouched(controlName: string): boolean {
+    return this.getControl(controlName).touched;
+  }
+
+  isInputValid(controlName: string, hasPattern = false): boolean {
+    if (controlName === 'email' && hasPattern) {
```

Comment:
for not duplicating the control names like here,
move it to `const enum` and use enum for these fields.
and you can use this enum in the place of initializing the form (ngOnInit)